nodejs/src/attachments.ts

```diff
@@ -0,0 +1,89 @@
+import { Attachment, AttachmentContentFormat, Kind } from '@kapeta/schemas';
+import * as Path from 'node:path';
+import * as FS from 'node:fs/promises';
+import { pack, unpack } from './utils';
+
+export function getAttachment(kind: Kind, filename: string) {
+    if (!kind.attachments) {
+        return null;
+    }
+    return kind.attachments.find((a) => a.filename === filename) || null;
+}
+
+export function setAttachment(kind: Kind, attachment: Attachment) {
+    if (!kind.attachments) {
+        kind.attachments = [];
+    }
+
+    const ix = kind.attachments.findIndex((a) => a.filename === attachment.filename);
+    if (ix >= 0) {
+        kind.attachments[ix] = attachment;
+        return;
+    } else {
+        kind.attachments.push(attachment);
+    }
+}
+
+export async function getAttachmentContent(kind: Kind, filename: string) {
+    const attachment = getAttachment(kind, filename);
+    if (!attachment) {
+        return null;
+    }
+    return readAttachmentContent(attachment.content.format, attachment.content.value);
+}
+
+export async function readAttachmentContent(format: AttachmentContentFormat, value: string) {
+    switch (format) {
+        case AttachmentContentFormat.Base64:
+            return Buffer.from(value, 'base64');
+        case AttachmentContentFormat.Plain:
+            return Buffer.from(value, 'utf-8');
+        case AttachmentContentFormat.Base64Gzip:
+            return unpack(value);
+        case AttachmentContentFormat.URL:
+            return fetch(value)
```

Comment:
Is this one potentially dangerous? 🤔  Or where does the value come from?

Reply:
It's basically user defined - but we dont evaluate anything coming from there.